### DISCUSSION TITLE
How does UsePersistentStore relate to AdoJobStore, is it the same thing?

### DISCUSSION THREAD

**Discussion by s3YwCf2ZbfJG4SHAfjQMAjtsf** [quartznet/quartznet#1779] [Q&A] [answered]
I am following the example defined here, without a configuration file (I think defaults are fine)
https://www.quartz-scheduler.net/documentation/quartz-3.x/quick-start.html#configuration

Am I misunderstanding?  It suggested I need to use **AdoJobStore**, but my code doesn't mention that.
https://www.quartz-scheduler.net/documentation/quartz-3.x/tutorial/advanced-enterprise-features.html#clustering

**UseClustering**, **UsePersistentStore**, **UseOracle** is not clearly defined in the documentation so other than the API help, I don't have much to go on.

I have only implemented my dev machine as a single instance against my Oracle DB (first day working with Quartz).

If I use this same app, configuration, and specific Oracle schema from another machine, is clustering configured correctly?

Thanks!   (So far VERY nice product, and it runs my job)

```
                _scheduler = await SchedulerBuilder.Create(properties)
                    // default max concurrency is 1
                    .UseDefaultThreadPool(x => x.MaxConcurrency = 1)
                    // this is the default 
                    // .WithMisfireThreshold(TimeSpan.FromSeconds(60))
                    .UsePersistentStore(x =>
                    {
                        // force job data map values to be considered as strings
                        // prevents nasty surprises if object is accidentally serialized and then 
                        // serialization format breaks, defaults to false
                        x.UseProperties = true;
                        x.UseClustering();
                        // there are other SQL providers supported too 
                        x.UseOracle($"Data Source=(deleted)");
                        // this requires Quartz.Serialization.Json NuGet package
                        x.UseJsonSerializer();
                    })
                    .BuildScheduler();
```

**Comment by lahma** [comment, marked as answer]
Yes, unfortunately we currently have the documentation following the "old configuration way" which exposes more types and configuration variables. You are using the fluent version (great!) which tries to hide unnecessary complexity and hopefully still allowing you to configure the complex parts too if needed.

In this case I'd advice to check the source code, many IDEs allow you to dive in by going into definition, it'll reveal what configuration will used, in this case:

https://github.com/quartznet/quartznet/blob/ab72bf7838accd208612bd2f0333c904a9d2602b/src/Quartz/SchedulerBuilder.cs#L115-L118

So it'll use `JobStoreTX` which is an `AdoJobStore` implementation.

Many of the fluent/c# configuration methods eventually set a property in the properties map - it's a strongly typed version protecting from typos.

**Reply by s3YwCf2ZbfJG4SHAfjQMAjtsf** [reply]
Thanks, I was just curious overall and especially whether I actually have written it to use "Clustering".  :)

**Reply by lahma** [reply]
Yes, with debugger you probably can catch the final configuration and see what configuration properties were set and with which values. But I think you're good with the clustering 🙂
